.gitlab/pipelines/stages/deploy/wait_kubecf.sh

```diff
@@ -9,51 +9,38 @@ source "${workspace}/.gitlab/pipelines/runtime/config.sh"
 # shellcheck disable=SC1090
 source "${workspace}/.gitlab/pipelines/runtime/binaries.sh"
 
-wait_kubecf() {
-  timeout 900 bash <<EOF
-while true; do
-  if '${KUBECTL}' get pods --namespace ${KUBECF_NAMESPACE} --no-headers --ignore-not-found | grep "router"; then
-    exit 0
-  fi
+echo "Waiting for the kubecf pods to be ready..."
+
+secret_name="${KUBECF_INSTALL_NAME}.with-ops"
+
+until "${KUBECTL}" get secret "${secret_name}" \
+  --namespace "${KUBECF_NAMESPACE}" \
+  --output jsonpath='{ .data.manifest\.yaml }' \
+  1> /dev/null 2> /dev/null; do
   sleep 1
 done
-EOF
-  timeout 900 bash <<EOF
-pod_count() {
-  '${KUBECTL}' get pods --namespace ${KUBECF_NAMESPACE} --no-headers --ignore-not-found \
-    | wc --lines
-}
-
-ready_pod_count() {
-  '${KUBECTL}' get pods --namespace ${KUBECF_NAMESPACE} --no-headers --ignore-not-found \
-    | awk '{ printf "%s-%s\n", \$2, \$3 }' | grep --line-regexp '\(.*\)\/\1-Running' \
-    | wc --lines
-}
-
-print_status() {
-  timestamp="\$(date +"%T")"
-  printf "%s - %d of %d pods ready.\n" "\${timestamp}" "\${1}" "\${2}"
-}
-
-i=0
-while true; do
-  total="\$(pod_count)"
-  ready="\$(ready_pod_count)"
-  if [[ "\${total}" > "0" ]] && [[ "\${total}" == "\${ready}" ]]; then
-    print_status "\${ready}" "\${total}"
-    exit 0
-  fi
-  if (( i % 60 == 0 )); then
-    print_status "\${ready}" "\${total}"
-  fi
-  sleep 1
-  i=\$((i + 1))
+
+instance_groups=$(
+  "${KUBECTL}" get secret "${secret_name}" \
+    --namespace "${KUBECF_NAMESPACE}" \
+    --output jsonpath='{ .data.manifest\.yaml }' \
+    | base64 --decode \
+    | "${YAML2JSON}" \
+    | "${JQ}" -r '.instance_groups[] | select(.lifecycle != "errand") | select (.lifecycle != "auto-errand") | .name'
+)
+
+for instance_group in $instance_groups; do
```

Comment:
Not that we _really_ need to worry about spaces in instance group names, but with the previous array change…
```suggestion
for instance_group in "${instance_groups[@]}"; do
```

Reply:
Fixed in https://github.com/SUSE/kubecf/pull/208/commits/1aaf9fa662e627f8f06c04e47543198503290895.